Fix full bank statement field description

lovo-h opened the pull request:
<!--
Did I add a title? A descriptive, yet concise, title.
-->

<!--
Issue: Link to the GitHub issue this PR addresses (if appropriate).
-->

Fixes #2180

## Changes proposed in this Pull Request:

<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->

<!--
Questions for the PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

This PR helps clarify what input is valid in the **Full bank statement** field. In order to convey this info, a tooltip was placed next to the input. This approach was selected because it was cleaner than adding more text in the description below the input field.

Because this was a minor UI update, no tests were added to this PR.

![image](https://user-images.githubusercontent.com/6844516/149678120-f63e6b90-4c8a-461e-a03d-9ebe873bca89.png)

## Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Please follow the following guidelines when writing testing instructions:

- Include screenshots if there is no similar flow in the critical flows: https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Critical-flows
- Assume instructions will be copied over to the Release Testing Instructions wiki page: https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions
- Assume instructions will be followed by external testers.
- Assume tester does not have intimate knowledge of Stripe.
-->

1. As a merchant, go to `WooCommerce > Settings > Payments > Stripe` and click on the `Settings` tab (at the top).
2. Scroll down to `Payments & transactions > Customer bank statement`.
3. Hover over the **question mark** icon, next to the input field of **Full bank statement.**
4. Ensure the following text is displayed: `The bank statement must contain only Latin characters, be between 5 and 22 characters, and not contain any of the special characters: \' " * < >`.

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [x] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)

lovo-h committed:
Fix typo: spacing between words and the dash line

lovo-h committed:
Add tooltip to bank statement field, indicating acceptable input

The tooltip indicates what input is acceptable in the bank statement
field.

lovo-h committed:
Update changelog: Fixed full bank statement field description

ricardo commented on client/settings/payments-and-transactions-section/index.js:
```diff
@@ -17,6 +19,19 @@ import {
+			'woocommerce-gateway-stripe'
+		) }
+	>
+		<span style={ { flexShrink: 0 } }>
```
Do we need the `flexShrink` as the parent is not a flex container?

lovo-h replied:
Good find. It was not needed.

ricardo reviewed (approved):
LGTM - pre-approving.

Just need to change the tooltip and notice copy as discussed via DM.

lovo-h committed:
Rephrase bank statement input requirement messages

Rephrasing in order to clarify what characters can be input for the
bank statement field.

lovo-h committed:
Remove unneeded style: flex-shrink
